### PR TITLE
Custom Auth Polish

### PR DESCRIPTION
**Pull request by bretambrose**
* Update custom auth docmentation
* Provide fallback URI encoding of signatures if not done by user

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java [hunk 1]
```diff
@@ -644,6 +646,14 @@ public AwsIotMqttConnectionBuilder withCustomAuthorizer(String username, String
...
         if (authorizerSignature != null)
         {
+            if (!authorizerSignature.contains("%")) {
+                try {
+                    authorizerSignature = URLEncoder.encode(authorizerSignature, StandardCharsets.UTF_8.toString());
+                } catch (UnsupportedEncodingException uee) {
+                    throw new CrtRuntimeException(uee.toString());
+                }
+            }
+
```
Comment (sfod) [inline sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java:656]: Similar to aws-crt-nodejs, we can check for a case when `tokenKeyName` and `tokenValue` are set while `authorizerSignature` is empty.

Comment (bretambrose) [reply sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java:663]: Done

File: sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqtt5ClientBuilder.java [hunk 2]
```diff
@@ -885,9 +888,16 @@ private String buildMqtt5FinalUsername(MqttConnectCustomAuthConfig config) throw
             }
 
             addToUsernameParam(paramList, "x-amz-customauthorizer-name", config.authorizerName);
+            
             if (usingSigning == true) {
                 addToUsernameParam(paramList, config.tokenKeyName, config.tokenValue);
-                addToUsernameParam(paramList, "x-amz-customauthorizer-signature", config.tokenSignature);
+
+                String encodedSignature = config.tokenSignature;
+                if (!encodedSignature.contains("%")) {
```
Comment (sbSteveK) [inline sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqtt5ClientBuilder.java:896]: Trivial: Should we log that we are doing this for debugging purposes?

File: sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java [hunk 3]
```diff
@@ -642,23 +650,21 @@ public AwsIotMqttConnectionBuilder withCustomAuthorizer(String username, String
...
-            if (tokenKeyName == null || tokenValue == null) {
-                Log.log(
-                    LogLevel.Warn,
-                    LogSubject.MqttClient,
-                    "Signed custom authorizers with signature will not work without a token key name and " +
-                    "token value. Your connection may be rejected/stalled on the IoT Core side due to this. Please " +
-                    "use the non-deprecate API and pass both the token key name and token value to connect to a " +
-                    "signed custom authorizer");
+            if (!authorizerSignature.contains("%")) {
+                try {
```
Comment (sbSteveK) [inline sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java:657]: Trival: Same as above, should we log that we have encoded the signature?

**Review by sbSteveK — APPROVED** [review]

**Review by sfod — APPROVED** [review]

**Pull request merged**